lib/segment/src/index/query_optimization/nested_filter.rs

```diff
@@ -0,0 +1,522 @@
+                Some(values) => {
+                    let len = values.len();
+                    let matching = values
+                        .iter()
+                        .copied()
+                        .positions(|i| range.check_range(i as FloatPayloadType))
+                        .collect();
+                    (matching, len)
+                }
+            }
+        })),
+        FieldIndex::FloatIndex(num_index) => Some(Box::new(move |point_id: PointOffsetType| {
+            match num_index.get_values(point_id) {
+                None => (vec![], 0),
+                Some(values) => {
+                    let len = values.len();
+                    let matching = values
+                        .iter()
+                        .copied()
+                        .positions(|i| range.check_range(i))
+                        .collect();
+                    (matching, len)
+                }
+            }
+        })),
+        _ => None,
+    }
+}
+
+pub fn get_nested_match_checkers(
+    index: &FieldIndex,
+    cond_match: Match,
+) -> Option<NestedMatchingIndicesFn> {
+    match cond_match {
+        Match::Value(MatchValue {
+            value: value_variant,
+        }) => match (value_variant, index) {
+            (ValueVariants::Keyword(keyword), FieldIndex::KeywordIndex(index)) => {
+                Some(Box::new(move |point_id: PointOffsetType| {
+                    match index.get_values(point_id) {
+                        None => (vec![], 0),
+                        Some(values) => {
+                            let len = values.len();
+                            let matching = values.iter().positions(|k| k == &keyword).collect();
+                            (matching, len)
+                        }
+                    }
+                }))
+            }
+            (ValueVariants::Integer(value), FieldIndex::IntMapIndex(index)) => {
+                Some(Box::new(move |point_id: PointOffsetType| {
+                    match index.get_values(point_id) {
+                        None => (vec![], 0),
+                        Some(values) => {
+                            let len = values.len();
+                            let matching = values.iter().positions(|i| i == &value).collect();
+                            (matching, len)
+                        }
+                    }
+                }))
+            }
+            _ => None,
+        },
+        Match::Text(MatchText { text }) => match index {
+            FieldIndex::FullTextIndex(full_text_index) => {
+                let parsed_query = full_text_index.parse_query(&text);
+                Some(Box::new(
+                    move |point_id: PointOffsetType| match full_text_index.get_doc(point_id) {
+                        None => (vec![], 0),
+                        Some(doc) => {
+                            let res = parsed_query.check_match(doc);
+                            // Not sure it is entirely correct
```

Comment:
Could you please elaborate?

Reply:
Unlike other types of indexes, in case of a full text search, we dot have an array of values to work with.
Either the indexed document matches the query or it does not.

If it matches, the result represents is a synthetic match on the first element to honor the interface.